### DISCUSSION TITLE
Migrate app from Provider to Riverpod

### DISCUSSION THREAD

**Discussion by chetferry** [rrousselGit/riverpod#443] [Q&A] [answered]
First let me say thank you so much for doing Riverpod.  I think it is awesome.

I am helping with an app that uses Provider.  We would like to switch to Riverpod.  We have about 8 providers and are doing the MultiProvider thing and also the ProxyProvider thing.  We believe that it would be a good move and help our app scale.  

Is it possible to bring in Riverpod incrementally?  Can Provider and Riverpod both be used in the same app, or do we need to make the switch all at once?  Thanks very much.

**Comment by rrousselGit** [comment, marked as answer]
Hello!

Could you explain what the difficulty is?

**Reply by chetferry** [reply]
Can I use legacy Providers and Riverpod providers in the same app?  We are going to migrate to Riverpod.  Just does it need to be done all at once?  No difficulties, just a preemptive question before we start the process.  Thanks.

**Reply by rrousselGit** [reply]
Sure you can. What's the issue?

**Reply by chetferry** [reply]
No issues.  Thanks for answering the question.
